artiq/coredevice/comm_analyzer.py

```diff
@@ -107,26 +107,35 @@ def decode_dump(data):
     data = data[1:]
     # only header is device endian
     # messages are big endian
-    parts = struct.unpack(endian + "IQbbb", data[:15])
-    (sent_bytes, total_byte_count,
-     error_occurred, log_channel, dds_onehot_sel) = parts
+    payload_length = struct.unpack(endian + "I", data[:4])[0]
```

Comment:
The way the `AnalyzerProxyReceiver` PR (#2300) is currently set up, we do endianness and extract `payload_length` in the receiver code:
```
                payload_length_word = await self.reader.readexactly(4)
                payload_length = struct.unpack(endian + "I", payload_length_word)[0]
                if payload_length > 10 * 512 * 1024:
                    # 10x buffer size of firmware
                    raise ValueError
                data = await self.reader.readexactly(payload_length + 11)
                self.receive_cb(endian, payload_length, data)
```
By leaving this stuff out here, I can call `decode_dump_header` directly in the callback function passed to `AnalyzerProxyReceiver`. Open to any suggestions on better structuring this.

Reply:
> Why would you want to do that? The proxy doesn't process the data and just passes it verbatim to the clients, no? 

Yes, and I would like to pass the data verbatim to the `receive_cb` as well but I'm not sure how to do this without copying the remaining data (through some concat or similar). That's why I pass the endian, payload_length and remaining data to the `receive_cb`. If I could pass the data verbatim, this PR would not be necessary as `decode_dump` could be called directly.

> It would be OK if the proxy repeats the endiannness bit and everything, keeping the exact same protocol everywhere is a plus.

Just to clarify, is this a suggestion for the proxy to send the endian + payload_length twice to clients (first 5 bytes)? If so I'll work on a PR for the proxy to implement this.

Reply:
> Why would it be sent twice? The suggestion is to use the exact same data/protocol that is sent by the devices.

Okay understood. Sorry about the confusion.

> I think some copy is unavoidable if we do this. There is some copying going on already in readexactly() anyway.

Okay, then I will close this PR and make modifications to the #2300 PR to ensure its the same data being passed along verbatim.